src/app/css-variables.ts

```diff
@@ -1,56 +1,88 @@
-    }
-    if (
-      currentState.charColMobile !== nextState.charColMobile &&
-      // this check is needed so on start up/load this doesn't override the value set above on "normal" mode.
-      isPhonePortraitSelector(state)
-    ) {
-      setCSSVariable('--tiles-per-char-column', nextState.charColMobile);
-    }
+export function createThemeObserver(): StoreObserver<{ theme: string; isPhonePortrait: boolean }> {
+  return {
+    id: 'theme-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      theme: settingsSelector(rs).theme,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ previous, current }) => {
+      // TODO (ryan): Hoist the feature flag check up to where the observer is created
+      // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
+      if ($featureFlags.themePicker && previous.theme !== current.theme) {
+        const themeClass = `theme-${current.theme}`;
+        document.body.className = themeClass;
+        syncThemeColor(current.isPhonePortrait);
+      }
+    },
+  };
+}
 
-    // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
-    if ($featureFlags.themePicker && currentState.theme !== nextState.theme) {
-      const themeClass = `theme-${nextState.theme}`;
-      document.body.className = themeClass;
-      syncThemeColor(isPhonePortraitSelector(state));
-    }
-  });
+export function createTilesPerCharColumnObserver(): StoreObserver<{
+  charCol: number;
+  charColMobile: number;
+  isPhonePortrait: boolean;
+}> {
+  return {
+    id: 'tiles-per-char-column-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      charCol: settingsSelector(rs).charCol,
+      charColMobile: settingsSelector(rs).charColMobile,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ previous, current }) => {
+      // Changes in phone landscape/portrait state
+      if (previous.isPhonePortrait !== current.isPhonePortrait) {
+        setCSSVariable(
+          '--tiles-per-char-column',
+          current.isPhonePortrait ? current.charColMobile : current.charCol,
+        );
+        syncThemeColor(current.isPhonePortrait);
+      }
 
-  // a subscribe on isPhonePortrait is needed when the user on mobile changes from portrait to landscape
-  // or a user on desktop shrinks the browser window below isphoneportrait threshold value
-  observeStore(isPhonePortraitSelector, (_prev, isPhonePortrait, state) => {
-    const settings = settingsSelector(state);
-    setCSSVariable(
-      '--tiles-per-char-column',
-      isPhonePortrait ? settings.charColMobile : settings.charCol,
-    );
-    syncThemeColor(isPhonePortrait);
-  });
+      if (previous.charCol !== current.charCol && previous.isPhonePortrait) {
+        setCSSVariable('--tiles-per-char-column', current.charCol);
+      }
```

Comment:
How necessary are these guards? I feel like these three could be replaced with a single

```
       setCSSVariable(
          '--tiles-per-char-column',
          current.isPhonePortrait ? current.charColMobile : current.charCol,
        );
```

which just gets run whenever the observer middleware triggers it. I don't think there would be any performance concerns with css variable updates that set exactly the same value, but I don't know enough about global css variables to have a strong opinion.

Reply:
I decided to just do this, left a comment on the new code.